lib/public/views/Runs/RunPerDataPass/RunsPerDataPassOverviewPage.js

```diff
@@ -247,17 +247,28 @@ export const RunsPerDataPassOverviewPage = ({
                                         dataPassId,
                                     },
                                 ),
-                                sessionService.hasAccess([BkpRoles.DPG_ASYNC_QC_ADMIN]) && h(
-                                    'button.btn.btn-danger',
-                                    {
-                                        onclick: () => {
-                                            if (confirm('Are you sure you want to delete ALL the QC flags for this data pass?')) {
-                                                perDataPassOverviewModel.discardAllQcFlags();
-                                            }
+                                sessionService.hasAccess([BkpRoles.DPG_ASYNC_QC_ADMIN]) && [
+                                    h(
+                                        'button.btn.btn-danger',
+                                        {
+                                            onclick: () => dataPass.frozen
+                                                ? perDataPassOverviewModel.unfreezeDataPass()
+                                                : perDataPassOverviewModel.freezeDataPass(),
                                         },
-                                    },
-                                    'Delete ALL QC flags',
-                                ),
+                                        `${dataPass.frozen ? 'Unfreeze' : 'Freeze'} the data pass`,
+                                    ),
+                                    !dataPass.frozen && h(
```

Comment:
I think it will be a bit confusing for the user not to see the delete button when the pass is frozen. What do you think about disabling the delete button with an informative message? Similar to for example the `+ QC Flag` button